### PR TITLE
Add compaction code w/o secret use

### PR DESCRIPTION
**Pull request by Joeavaikath**
Feature: Add complete compaction code
This code has the Ceph S3 credentials removed. 

Fix: Refactor secret name

### CODE REVIEW

File: kubeflow-pipelines/odh-ray-cluster/secret-generator.yaml [hunk 1]
```diff
@@ -2,6 +2,6 @@ apiVersion: viaduct.ai/v1
 kind: ksops
 metadata:
   # Specify a name
-  name: aws-secret-generator
+  name: ceph_S3-secret-generator
```
Comment (gmfrasca) [inline kubeflow-pipelines/odh-ray-cluster/secret-generator.yaml:5]: mixing between _ and - is visually jarring if uneeded fwiw

File: kubeflow-pipelines/pipeline-code/ray-code.py [hunk 2]
```diff
@@ -28,13 +28,68 @@ def initialize_ray():
...
+
+    # Read path
+    read_path = f'{prefixPathRead}/metric={metricName}'
+
+    # Write path
+    write_path = f'{prefixPathWrite}/metric={metricName}'
+
+    # Prerequisite - Create S3FileSystem in PyArrow
+    # Why: Allows us to specify a custom endpoint
+    AWS_ACCESS_KEY = "<Ceph S3 Creds Here>"
```
Comment (gmfrasca) [inline kubeflow-pipelines/pipeline-code/ray-code.py]: could you use `os.environ['AWS_ACCESS_KEY']` here?

Comment (Joeavaikath) [reply kubeflow-pipelines/pipeline-code/ray-code.py]: The image doesn't come with it preloaded, so I would have to set it from somewhere after the pod is created. 
I have the secret up too, if only I could spec the pipeline python code to grab it when the pod is created.

**Comment by Joeavaikath** [comment]
The S3 credentials can be passed as runtime parameters in Kubeflow Pipelines UI, latest commit adds that.

File: kubeflow-pipelines/odh-ray-cluster/secret-generator.yaml (hunk 1 above)
Comment (Joeavaikath) [reply kubeflow-pipelines/odh-ray-cluster/secret-generator.yaml:5]: secrets have a naming convention as well: only lowercase chars, ' - ' and ' . ' are allowed. Making the change

**Review by anishasthana — APPROVED** [review]
/lgtm

**Pull request closed without merging**